src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/ThrowHelper.cs

```diff
@@ -5,34 +5,50 @@
 using System.Security.Authentication;
 using System.Net.Security;
 using static Microsoft.Quic.MsQuic;
+using System.Diagnostics.CodeAnalysis;
 
 namespace System.Net.Quic;
 
 internal static class ThrowHelper
 {
     internal static QuicException GetConnectionAbortedException(long errorCode)
     {
-        return errorCode switch
-        {
-            -1 => GetOperationAbortedException(), // Shutdown initiated by us.
-            long err => new QuicException(QuicError.ConnectionAborted, err, SR.Format(SR.net_quic_connectionaborted, err)) // Shutdown initiated by peer.
-        };
+        return new QuicException(QuicError.ConnectionAborted, errorCode, SR.Format(SR.net_quic_connectionaborted, errorCode));
```

Comment:
I would prefer an assert here so that we don't accidentally call this with the old "unset" value of -1, same for the other method